src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/components/step_index_pattern/step_index_pattern.js

```diff
@@ -77,6 +77,10 @@ export class StepIndexPattern extends Component {
 
     if (query.endsWith('*')) {
       const exactMatchedIndices = await ensureMinimumTime(getIndices(esService, query, MAX_SEARCH_SIZE));
+      // If the search changed, discard this state
+      if (query !== this.state.query) {
+        return;
```

Comment:
This is probably a common enough problem that we might want to come up with a systematic way to solve it. In past systems, I've used a request queue. We usually had an interrupt mechanism, too, which would allow aborting the current request via throwing an exception in the currently running request, and then we'd start the new request. Anyway. Not probably worth it for this one case, but something to think about if this is systemic (and I suspect it is).

Reply:
Yea definitely. The one downside to this is we're not actively throwing out older requests but just discarding the results so there is room for optimization. I'll create a ticket to track the effort to improve this or figure out how it fits into some common module.

Reply:
Created #17150